Sources/Sentry/SentrySpan.m

```diff
@@ -84,10 +96,50 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
         _spanId = context.spanId;
         _sampled = context.sampled;
         _origin = context.origin;
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        if (SentrySDK.options.enableContinuousProfiling) {
+            _profileSessionIDs = [NSMutableSet<NSString *> set];
+            if (SentryContinuousProfiler.isCurrentlyProfiling) {
+                [_profileSessionIDs
+                    addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+            }
+            [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+                addObserver:self
+                   selector:@selector(linkProfiler)
+                       name:kSentryNotificationContinuousProfileStarted];
+            [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+                addObserver:self
+                   selector:@selector(linkProfiler)
+                       name:kSentryNotificationContinuousProfileStopped];
```

Comment:
`m`: I guess using the NSNotificationCenter is fine for an experimental release, but I believe this could become a performance problem when you have hundreds of spans or more running simultaneously. I'm unsure how performant the NSNotifcationCenter is when it has to loop through.

Reply:
yeah personally i don't love it either, this just got it up and running the quickest. I can move it to my preferred pattern of delegation. less spaghetti like.

Reply:
Honestly if the question is just one of performance, I don't know if we're going to be able to beat it by just iterating through an array of listener instances 🤔 It's fair to assume that Apple has sunk lots of resources into optimizing such a widely used API.